### PR TITLE
Update how collection instruments get loaded for BRES

### PR DESCRIPTION
**Pull request by ryangrundy7**
## Motivation and Context
The purpose of this PR is to update how collection instruments get loaded for BRES. At the time this was created, the collection exercise was hardcoded so needed to be updated and other parts of the code needed refactoring to tidy it up. After this goes in:

- The Jenkins job will need to be updated for the new environment variables,
- The asymmetric keys that are they need to be checked in case they're out of date

## What has changed

- The collection exercise has now been updated to an environment variable
- Added more environment variables
- Added a config so all environment variables can be put in one place
- Added a Dockerfile for it to run

## How to test?
- Make sure that the tests pass 
- try attempting to run the entire service with sdx-seft-publisher-service

At the moment, you'll have to build the image locally to get it to work so do `docker build -t sdcplatform/ras-rabbit-adaptor-service .` to build the image. You'll have to do the same with [sdx-seft-publisher-service]( https://github.com/ONSdigital/sdx-seft-publisher-service) 

Run the Docker-compose file on `sdc-ci-upload-compose` with the image and it should all start up correctly. To do an entire run through of the uploading a collection instrument, You may have to edit code in `ras-rabbit-adaptor`  by turning off the decrypter. 

Other repos used: 
[sdx-seft-publisher-service](https://github.com/ONSdigital/sdx-seft-publisher-service)
[sdc-ci-upload-compose](https://github.com/ONSdigital/sdc-ci-upload-compose)

### CODE REVIEW

File: Pipfile [hunk 1]
```diff
@@ -12,11 +12,11 @@ python_version = "3.6"
...
 [packages]
 
-cryptography = "==1.9"
-PyYAML = "==3.12"
+cryptography = "==2.3"
+PyYAML = "==3.13"
 requests = "==2.18.2"
 sdc-rabbit = "==0.3.4"
-sdc-cryptography = "==0.1.6"
+sdc-cryptography = "==0.2.3"
```
Comment (shahi645) [inline Pipfile:19]: 👍

**Review by shahi645 — APPROVED** [review]
These changes look good to me, works for me locally. Nice refactor so far!

**Comment by shahi645** [comment]
@ryangrundy7 may want to introduce some checks to this repo, like codacy and codecov

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+WORKDIR /app
+COPY . /app
+COPY app/tests /app/tests
+COPY /app/tests/secrets.json /app/tests
+EXPOSE 8080
+RUN pip3 install -U pipenv && pipenv install --system --deploy
+
+ENTRYPOINT ["python3"]
+CMD ["main.py"]
```
Comment (shahi645) [inline Dockerfile]: @ryangrundy7 FYI can just do `CMD  ["python3","main.py"]` and remove line `10` to save some code, but only very minor thing

File: main.py [hunk 3]
```diff
@@ -44,7 +43,7 @@ def main():
         exchange=os.getenv('RABBIT_EXCHANGE', 'test'),
```
Comment (LJBabbage) [inline main.py:43]: Any reason you didn't put these in the config like everything else?

**Review by LJBabbage — APPROVED** [review]

File: main.py (hunk 3 above)
Comment (shahi645) [reply main.py:43]: Will leave as this is the only place these get called

**Pull request merged**